pkg/apiclient/apiclient.go

```diff
@@ -73,7 +73,7 @@ func NewClientFromOpts(opts Opts) (context.Context, Client, error) {
 		if opts.AuthSupplier == nil {
 			return nil, nil, fmt.Errorf("AuthSupplier cannot be empty when connecting to Argo Server")
 		}
-		return newHTTP1Client(opts.ArgoServerOpts.GetURL(), opts.AuthSupplier(), opts.ArgoServerOpts.InsecureSkipVerify, opts.ArgoServerOpts.Headers, opts.ArgoServerOpts.HTTP1Client)
+		return newHTTP1Client(opts.ArgoServerOpts.GetURL(), opts.AuthSupplier(), opts.ArgoServerOpts.InsecureSkipVerify, opts.ArgoServerOpts.Headers, opts.ArgoServerOpts.HTTP1Client, opts.ArgoServerOpts.ClientCert, opts.ArgoServerOpts.ClientKey)
```

Comment:
this is fine for now, but given how many options we're passing through, we may want to consider passing through `ArgoServerOpts` entirely (or `opts` entirely)

Reply:
Nice catch, I can refactor this in a commit here will improve readability by a lot.

Reply:
I am passing opts to the `newHTTP1Client`